sqlserver/datadog_checks/sqlserver/connection.py

```diff
@@ -432,6 +434,31 @@ def _get_access_info(self, db_key, db_name=None):
                 driver = self.DEFAULT_DRIVER
         return dsn, host, username, password, database, driver
 
+    def _get_host_with_port(self):
+        """Return a string with format host,port.
+        If the host string in the config contains a port, that port is used.
+        If not, any port provided as a separate port config option is used.
+        If the port is misconfigured or missing, default port is used.
+        """
+        host = self.instance.get('host')
+        if not host:
+            return None
```

Comment:
do we log a warning at all when `host` is not set in the instance config?

Reply:
i can add a warning

Reply:
added warning messages for a few defaults in a separate commit

Reply:
Reposting from the offline discussion: I think we shouldn't log warnings for fallbacks which are considered safe & normal. It could be spammy for many customers and make them think something is wrong. We should only log warnings when something wrong is happening that the customer should fix.

Reply:
we should log when the `host` isn't set though

Reply:
Why? There could be people relying on that default behavior of falling back to localhost. Should they start seeing warnings in their logs now? 

Seems OK to default to localhost. Many tools do that anyway, like commandline db clients which connect to localhost when invoked directly on the same host as the DB process.

Reply:
hmmm alright 🤷 maybe it only makes sense to log it, if we get `Conn Refused` on localhost or something.